README.md

```diff
@@ -4,60 +4,143 @@
+value every contributor has added to the project. SourceCred then enables the
+project to issue tokens, called Grain, to contributors based on their Cred.
+Grain is purchased by sponsors of the project, and gives sponsors the ability
+to influence Cred scores.
 
-Cred is computed by constructing a Cred Graph showing every contribution to a project, and then running a version of the PageRank
-algorithm to assign scores to every contribution and contributor.
+You can read more at [sourcecred.io].
 
-We currently support loading data from GitHub, Discourse, Discord, and via custom "initiatives".
+[sourcecred.io]: https://sourcecred.io/
 
-Please check out [our website] for more information. If you'd like to get involved as a contributor, please drop by [our Discord]
-and say "hi"!
+## Plugin Architecture
 
-[our website]: https://sourcecred.io/
-[our Discord]: https://sourcecred.io/discord
+SourceCred is organized around a plugin architecture, which ensures that it can
+track and reward any kind of contribution, so long as you can assign addresses
+to contributions, and record how they relate to one another. Currently, we have the following four plugins:
+
+- **GitHub**: Loads GitHub repositories, including issues, pull requests, and reviews
+- **Discourse**: Loads Discourse forums, including posts, topics, and likes
+- **Discord**: Loads Discord servers, including messages and reactions
+- **Initiatives**: Loads manually added contributions. Still in alpha.
+
+Every plugin has a two-part name in the form `$OWNER/$NAME`; for example,
+SourceCred's own GitHub plugin is named `sourcecred/github`.
+
+## The Instance System
+
+SourceCred is organized around the concept of "instances". A SourceCred instance
+contains all of the configuration and data associataed with Cred and Grain, and
+optionally may be set up as a deployable website that displays those scores.
+Each instance has the following directory structure:
+
+```
+./sourcecred.json   # Lists enabled plugins
+./config            # User-edited config files
+./data              # Persistent data, e.g. ledger history
+./output            # Output data, may be removed/regenerated
+./site              # Bundled frontend, if included
+./cache             # Temporary data, should not be checked in to git
+```
 
-## Current Status
+We recommend storing instances in a Git repository. The best way to set up an
+instance is by forking [sourcecred/example-instance].
 
-SourceCred is still in beta; as such, the interfaces are in flux and the documentation is spotty.
-We're working on a polished release which will include more documentation, and more maintainable instances. We expect this to land
-by early July.
+Once your instance is setup, you can update it with the following commands:
 
-For now, if you want to use SourceCred, you might start by forking [MetaGame's Cred Instance].
+- `sourcecred load`: Regenerate the cache
+- `sourcecred graph`: Recompute graphs from cache
```

Comment:
Wondering why I might want to recompute graph from cache? Because it will only update the existing cached data, saving download time? If not doing that, why not just run score and re-run Cred calculations?

Reply:
Yeah, I'm considering merging the graph and score commands to make this simpler. But the basic answer is that both `load` and `graph` can load individual plugins, so suppose that you have two plugins, `sourcecred/fast` and `sourcecred/slow`. Slow takes forever to load, fast is fast, and you got new data in fast.

Then you can do:
`sourcecred load sourcecred/fast`
`sourcecred graph sourcecred/fast`
`sourcecred score`

And now you have up-to-date data from fast, and didnt need to reload slow.

Updated the text here to explain that.